asgi_correlation_id/middleware.py

```diff
@@ -38,16 +46,24 @@ async def __call__(self, scope: 'Scope', receive: 'Receive', send: 'Send') -> No
             await self.app(scope, receive, send)
             return
 
+        # Try to load request ID from the request headers
         header_value = Headers(scope=scope).get(self.header_name.lower())
 
         if not header_value:
-            id_value = uuid4().hex
-        elif self.validate_header_as_uuid and not is_valid_uuid(header_value):
-            logger.warning('Generating new UUID, since header value \'%s\' is invalid', header_value)
-            id_value = uuid4().hex
+            # Generate request ID if none was found
+            id_value = self.generator()
+        elif self.validator and not self.validator(header_value):
+            # Also generate a request ID if one was found, but it was deemed invalid
+            logger.warning('Generating new request ID, since header value \'%s\' is invalid', header_value)
```

Comment:
Logging user input strings (especially for unauthenticated users) is what caused Log4j, and is now recommended to not be done unless debug is on. 
I'd either remove the header value and allow people to extend logging if they'd like by adding it to their own validator, or make it a debug argument. 

I did [this](https://github.com/snok/django-guid/blob/main/django_guid/utils.py#L29) in Django-GUID, mostly because I didn't want to remove a warning log that's been there forever(I know people have dashboards to monitor it). Since this is a major bump, I think we should. 😊